### PR TITLE
feat(atenlib): add ops(squeeze)

### PR DESCRIPTION
**Pull request by xiaowuhu**

### CODE REVIEW

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 1]
```diff
@@ -5053,10 +5053,21 @@ def aten_square(self: TensorType) -> TensorType:
...
 
 
-def aten_squeeze(self: TensorType) -> TensorType:
+@torch_op("aten::squeeze", trace_only=True)
+def aten_squeeze(self: TTensor, dim: Optional[int] = None) -> TTensor:
     """squeeze(Tensor(a) self) -> Tensor(a)"""
 
-    raise NotImplementedError()
+    rank = op.Size(op.Shape(self))
+    if rank == 0:
```
Comment (fatcat-z) [inline onnxscript/function_libs/torch_aten/ops/core.py:5061]: Do we still need this if dim is None?

File: onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py [hunk 2]
```diff
@@ -556,6 +557,13 @@ def _where_input_wrangler(
         matcher=lambda sample: len(sample.args[0]) == 0,
         reason="Empty perm is not supported",
     ),
+    skip(
+        "squeeze",
+        matcher=lambda sample: len(sample.args) > 0
+        and len(sample.input.shape) > 0
+        and sample.input.shape[sample.args[0]] != 1,
+        reason="Cannot select an axis to squeeze out which has size not equal to one",
```
Comment (fatcat-z) [inline onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:565]:
```suggestion
        reason="Cannot squeeze a dimension which has a value different than 1",
```

**Review by fatcat-z — APPROVED** [review]
LGTM, thanks.

File: onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py (hunk 2 above)
Comment (xiaowuhu) [reply onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:565]: the warning message is coping from onnxruntime. :)

**Comment by titaiwangms** [comment]
@xiaowuhu I think this op is breaking CI test. Could you take a look?

**Pull request merged**